finetuner/experiment.py

```diff
@@ -180,6 +181,82 @@ def create_run(
         )
         return run
 
+    def _prepare_data(
+        self,
+        train_data: Union[DocumentArray, str],
+        eval_data: Union[DocumentArray, str],
+        eval_callback: EvaluationCallback,
+        run_name: str,
+    ):
+        """Upload data to Hubble and returns their names.
+
+        Uploads all data needed for fine-tuning - training data,
+        evaluation data and query/index data for `EvaluationCallback`.
+
+        Checks not to upload same dataset twice.
+
+        :param train_data: Either a `DocumentArray` for training data or a
+            name of the `DocumentArray` that is pushed on Hubble.
+        :param train_data: Either a `DocumentArray` for evaluation data or a
+            name of the `DocumentArray` that is pushed on Hubble.
+        :param eval_callback: Evaluation callback that contains query and index data
+            and gets modified in-place.
+        :param run_name: Name of the run.
+        :return: Name(s) of the uploaded data.
+        """
+        train_data_name = push_data_to_hubble(
+            client=self._client,
+            data=train_data,
+            data_type=TRAIN_DATA,
+            experiment_name=self._name,
+            run_name=run_name,
+        )
+        eval_data_name = None
+        if eval_data:
+            if eval_data == train_data:
+                eval_data_name = train_data_name
+            else:
+                eval_data_name = push_data_to_hubble(
+                    client=self._client,
+                    data=eval_data,
+                    data_type=EVAL_DATA,
+                    experiment_name=self._name,
+                    run_name=run_name,
+                )
+        if eval_callback:
+            if not eval_callback.query_data:
+                eval_callback.query_data = (
+                    eval_data_name if eval_data_name else train_data_name
```

Comment:
Is the training data automatically used as eval data, if no eval data is provided?

Reply:
Yes, however I'm not sure if it makes sense. 
Do you think we should do something like this?
```python
query_data = eval_data if eval_data else None
```

Reply:
Oh, you asked a different question. You're right, that doesn't make sense.